File: backend/app/schemas/kds.py
"""KDS ticket schemas (M9).

``KdsTicketRead`` is a display DTO built explicitly in ``kds_service`` (it
flattens product + modifier names off the order lines), not a plain
``from_attributes`` projection.
"""

from datetime import datetime

from pydantic import BaseModel

from app.models.kds import KdsStatus, Station
from app.models.order import OrderChannel


class KdsLineRead(BaseModel):
    order_item_id: int
    product_id: int
    product_name: str
    qty: int
    modifiers: list[str]


class KdsTicketRead(BaseModel):
    id: int
    order_id: int
    order_number: str
    table_number: str | None
    channel: OrderChannel
    station: Station
    status: KdsStatus
    printed_at: datetime
    bumped_at: datetime | None
    lines: list[KdsLineRead]

File: frontend/src/types/kds.ts
import type { OrderChannel } from '@/types/order'
import type { Station } from '@/types/product'

export type KdsStatus = 'new' | 'in_progress' | 'done'

export interface KdsLine {
  order_item_id: number
  product_id: number
  product_name: string
  qty: number
  modifiers: string[]
}

/** A kitchen-display ticket (M9 KdsTicketRead). */
export interface KdsTicket {
  id: number
  order_id: number
  order_number: string
  table_number: string | null
  channel: OrderChannel
  station: Station
  status: KdsStatus
  printed_at: string
  bumped_at: string | null
  lines: KdsLine[]
}
